pkg/oci/oci.go

```diff
@@ -0,0 +1,326 @@
+}
+
+func GetHostString(repository string) (string, error) {
+	repo, err := reference.Parse(repository)
+	if err != nil {
+		return "", fmt.Errorf("parse repository %q: %w", repository, err)
+	}
+	if named, ok := repo.(reference.Named); ok {
+		return reference.Domain(named), nil
+	}
+	return "", fmt.Errorf("image has to be a named reference")
+}
+
+func SetupAuthVariablesAndFlags(cmd *cobra.Command, authOptions *AuthOptions) {
+	// Flag inspired by https://github.com/containers/common/blob/cac40138f7e3c2b29ca32e64348535516bf6aa51/pkg/auth/cli.go#L48
+	cmd.Flags().StringVar(&authOptions.AuthFile, "authfile", DefaultAuthFile,
+		"path of the authentication file. This overrides the REGISTRY_AUTH_FILE environment variable")
+	viper.BindPFlag("registry.auth_file", cmd.Flags().Lookup("authfile"))
+	viper.BindEnv("registry.auth_file", "REGISTRY_AUTH_FILE")
+
+	cmd.Flags().BoolVar(&authOptions.Insecure, "insecure", false, "allow connections to HTTP only registries")
+}
+
+func CreateAuthClient(repository string, authOptions *AuthOptions) (*oras_auth.Client, error) {
+	logrus.Debugf("Using auth file %q", authOptions.AuthFile)
+
+	var cfg *configfile.ConfigFile
+	var err error
+
+	// 1. Explicitly setting the auth file
+	// 2. Using the default auth file
+	// 3. Using the default docker auth file if 2. doesn't exist
+	if authOptions.AuthFile != DefaultAuthFile {
+		authFileReader, err := os.Open(authOptions.AuthFile)
+		if err != nil {
+			return nil, fmt.Errorf("open auth file %q: %w", authOptions.AuthFile, err)
+		}
+		defer authFileReader.Close()
+		cfg, err = config.LoadFromReader(authFileReader)
+		if err != nil {
+			return nil, fmt.Errorf("load auth config: %w", err)
+		}
+	} else if _, err := os.Stat(authOptions.AuthFile); err == nil {
+		authFileReader, err := os.Open(authOptions.AuthFile)
+		if err != nil {
+			return nil, fmt.Errorf("open auth file %q: %w", authOptions.AuthFile, err)
+		}
+		defer authFileReader.Close()
+		cfg, err = config.LoadFromReader(authFileReader)
+		if err != nil {
+			return nil, fmt.Errorf("load auth config: %w", err)
+		}
+	} else {
+		logrus.Debugf("Couldn't find default auth file %q...", authOptions.AuthFile)
+		logrus.Debugf("Using default docker auth file instead")
+		logrus.Debugf("$HOME: %q", os.Getenv("HOME"))
+
+		cfg, err = config.Load("")
+		if err != nil {
+			return nil, fmt.Errorf("load auth config: %w", err)
+		}
+	}
+
+	hostString, err := GetHostString(repository)
+	if err != nil {
+		return nil, fmt.Errorf("get host string: %w", err)
+	}
+	authConfig, err := cfg.GetAuthConfig(hostString)
+	if err != nil {
+		return nil, fmt.Errorf("get auth config: %w", err)
+	}
+
+	return &oras_auth.Client{
+		Credential: oras_auth.StaticCredential(hostString, oras_auth.Credential{
+			Username:    authConfig.Username,
+			Password:    authConfig.Password,
+			AccessToken: authConfig.Auth,
+		}),
+	}, nil
+}
+
+func CreateRemoteRepository(image string, authOpts *AuthOptions) (*remote.Repository, error) {
```

Comment:
I think I saw a comment on naming before in on another PR - this _sounds_ like it would create a new repo on a remote. Maybe just dropping the `Create` would be okay as it just defines a reference to the remote repo in that returned variable? `repo, _ := RemoteRepository(...)`?

Reply:
Xref https://github.com/inspektor-gadget/inspektor-gadget/pull/2058#discussion_r1332719548

I think calling it `NewRepository` and adding some documentation will clarify things. What do you think?